tests/StringValidatorTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Brick\Validation\Tests;
+
+use Brick\Validation\Validator\StringValidator;
+
+/**
+ * Unit tests for the String validator.
+ */
+class StringValidatorTest extends AbstractTestCase
+{
+    public function testStringValidator()
+    {
+        $validator = new StringValidator();
+
+        $this->doTestValidator($validator, [
+            ''   => [],
+            null => [],
+        ]);
+    }
+
+    public function testStringValidatorWithTooShortString()
+    {
+        $validator = new StringValidator(5, 0);
+
+        $this->doTestValidator($validator, [
+            '123' => ['validator.string.too-short'],
+        ]);
+    }
+
+    public function testStringValidatorWithTooLongString()
+    {
+        $validator = new StringValidator(5, 10);
+
+        $this->doTestValidator($validator, [
+            '12345678910' => ['validator.string.too-long'],
+        ]);
+    }
+
+    public function testStringValidatorWithNonUtf8String()
+    {
+        $validator = new StringValidator();
+
+        $this->doTestValidator($validator, [
+            file_get_contents(__DIR__ . '/non_utf8.txt') => ['validator.string.encoding'],
```

Comment:
Could you use an inline string here instead of a file, given its tiny size? Using the format `"\x00\x01..."`

Reply:
I remove the ```non_utf8.txt``` and I use the ```base64_encode``` function to encode the non-utf8 string firstly.
Then I use ```base64_decode``` function to decode string to let the method do validate.